packages/transaction-metrics-exporter/src/block-processor.ts

```
@@ -0,0 +1,266 @@
+          ],
+          jsonrpc: '2.0',
+          id: '2',
+        },
+        (err: any, result: any) => {
+          if (!err) {
+            result?.result
+              ?.filter?.(
+                ({ type, callType }: any) => type === 'create' || callType === 'delegatecall'
+              )
+              ?.forEach?.((data: any) =>
+                this.logEvent(LoggingCategory.InternalTransaction, {
+                  ...data,
+                  blockNumber,
+                  createdContractCode: undefined,
+                  init: undefined,
+                })
+              )
+          }
+        }
+      )
+
+      // tslint:disable-next-line
+      const labels = {
+        to: parsedTx ? tx.to : NOT_WHITELISTED_ADDRESS,
+        methodId: toMethodId(tx.input, parsedTx != null),
+        status: receipt.status.toString(),
+      } as labelValues
+
+      Counters.transaction.inc(labels)
+      Counters.transactionGasUsed.observe(labels, receipt.gasUsed)
+      Counters.transactionLogs.inc(labels, (receipt.logs || []).length)
+
+      if (parsedTx) {
+        Counters.parsedTransaction.inc({
+          contract: parsedTx.callDetails.contract,
+          function: parsedTx.callDetails.function,
+        })
+
+        this.logEvent(LoggingCategory.ParsedTransaction, {
+          ...parsedTx.callDetails,
+          hash: tx.hash,
+          blockNumber,
+        })
+        try {
+          for (const event of logExplorer.getKnownLogs(receipt)) {
+            Counters.transactionParsedLogs.inc({
+              contract: parsedTx.callDetails.contract,
+              function: parsedTx.callDetails.function,
+              log: event.event,
+            })
+
+            // @ts-ignore We want to rename event => eventName to avoid overwriting
+            event.eventName = event.event
+            delete event.event
+
+            this.logEvent(LoggingCategory.ParsedLog, event)
+          }
+          // tslint:disable-next-line
+        } catch {}
```

I don't think we should just silently fail here?